### PR TITLE
CON-999-support-user-banner-initial

### PR DESCRIPTION
**Pull request by Stephek**

### CODE REVIEW

File: src/SFA.DAS.EmployerAccounts.Web/Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -115,6 +115,7 @@
         @Html.Partial("_SVGSprite")
     </div>
     <header role="banner" id="global-header" class="with-proposition">
+        @SFA.DAS.EmployerAccounts.Web.Extensions.HtmlHelperExtensions.RenderSupportBanner(Html, Model);
```
Comment (dev4valtech) [inline src/SFA.DAS.EmployerAccounts.Web/Views/Shared/_Layout.cshtml]: Perhaps call child action directly using `@Html.RenderAction`

File: src/SFA.DAS.EmployerAccounts.Web/Views/Shared/_SupportUserBanner.cshtml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+</div>
+
+<div id="service-console-msg">
+
+    <div id="global-header-bar"></div>
+
+    <div class="header-wrapper">
+
+        <div class="content">
+            <p>You are viewing as @(Model.Account?.Name ?? string.Empty) (account ID:  @(Model.Account?.PublicHashedId ?? string.Empty))</p>
```
Comment (dev4valtech) [inline src/SFA.DAS.EmployerAccounts.Web/Views/Shared/_SupportUserBanner.cshtml]: Why are we checking for null here, are we expecting `Account` to not be populated? However we have safe checks in the controller action that will assure that we render this view only when applicable. Feels like I may have missed something here.

Comment (Stephek) [reply src/SFA.DAS.EmployerAccounts.Web/Views/Shared/_SupportUserBanner.cshtml]: When EmployerFinance uses this then the account will be null and an error would be thrown.  This partial shouldn't make assumptions about what calls it.

File: src/SFA.DAS.EmployerAccounts.Web/Controllers/BaseController.cs [hunk 3]
```diff
@@ -151,5 +152,15 @@ public void RemoveFlashMessageFromCookie()
         {
             _flashMessage.Delete(FlashMessageCookieName);
         }
+
+        public virtual ActionResult SupportUserBanner(IAccountIdentifier model = null)
```
[the remark this answers is not in the document]
Comment (Stephek) [reply src/SFA.DAS.EmployerAccounts.Web/Controllers/BaseController.cs]: It's not redundant but available for any override.  EmployerAccounts overrides and uses it, EmployerFinance doesn't.  This is the default implementation.

File: src/SFA.DAS.EmployerAccounts.Web/Controllers/EmployerTeamController.cs [hunk 4]
```diff
@@ -323,6 +321,33 @@ public async Task<ActionResult> HideWizard(string hashedAccountId)
...
+                    else
+                    {
+                        account = response.Data.Account;
+                    }
+                }
+
+                return PartialView("_SupportUserBanner", new SupportUserBannerViewModel() { Account = account });
+            }
+
+            return base.SupportUserBanner(model);
```
[the remark this answers is not in the document]
Comment (Stephek) [reply src/SFA.DAS.EmployerAccounts.Web/Controllers/EmployerTeamController.cs]: It is just to save having to implement it in the EmployerFinance site but I can change it if that is confusing you.

File: src/SFA.DAS.EmployerAccounts.Web/Extensions/HtmlHelperExtensions.cs [hunk 5]
```diff
@@ -22,5 +26,39 @@ public static MvcHtmlString CommaSeperatedAddressToHtml(this HtmlHelper htmlHelp
...
+        {
+            if(htmlHelper.IsSupportUser())
+            {
+                return htmlHelper.Action("SupportUserBanner", new { model = model.Data });
+            }
+
+            return MvcHtmlString.Empty;
+        }
+
+        public static bool IsSupportUser(this HtmlHelper htmlHelper)
```
[the remark this answers is not in the document]
Comment (Stephek) [reply src/SFA.DAS.EmployerAccounts.Web/Extensions/HtmlHelperExtensions.cs]: I don't think that would be any help but I can make it a private method of the controller.

File: src/SFA.DAS.EmployerAccounts.Web/Views/Shared/Empty.cshtml [hunk 6]
```diff
@@ -0,0 +1 @@
+﻿@model object
```
[the remark this answers is not in the document]
Comment (Stephek) [reply src/SFA.DAS.EmployerAccounts.Web/Views/Shared/Empty.cshtml]: Good point!

**Comment by cofaulco** [comment]
These changes were all merged into master as part of the feature branch, closing and deleting the branch

**Pull request closed without merging**